task/build-paketo-ubi-builder/0.1/build-paketo-ubi-builder.yaml

```diff
@@ -0,0 +1,527 @@
+            done
+            return $status
+          }
+          
+          ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+          if [ -f "$ca_bundle" ]; then
+            echo "INFO: Using mounted CA bundle: $ca_bundle"
+            cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+            update-ca-trust
+          fi
+          
+          echo "##########################################################################################"
+          echo "Pull the image from the OCI storage."
+          echo "##########################################################################################"
+          buildah --storage-driver $STORAGE_DRIVER pull $IMAGE
+          
+          echo "##########################################################################################"
+          echo "## Copy within the container of the image the sbom files"
+          echo "##########################################################################################"
+          #base_image_name=$(buildah inspect --format '{{ index .ImageAnnotations "org.opencontainers.image.base.name"}}' $IMAGE | cut -f1 -d'@')
+          #base_image_digest=$(buildah inspect --format '{{ index .ImageAnnotations "org.opencontainers.image.base.digest"}}' $IMAGE)
+          
+          container=$(buildah --storage-driver $STORAGE_DRIVER from --pull-never $IMAGE)
+          buildah --storage-driver $STORAGE_DRIVER copy $container sbom-cyclonedx.json sbom-purl.json /root/buildinfo/content_manifests/
+          
+          # The image build don't include "ImageAnnotations" as "null"
+          #buildah config -a org.opencontainers.image.base.name=${base_image_name} -a org.opencontainers.image.base.digest=${base_image_digest} $container
+          
+          BUILDAH_ARGS=()
+          if [ "${SQUASH}" == "true" ]; then
+            BUILDAH_ARGS+=("--squash")
+          fi
+          
+          buildah --storage-driver $STORAGE_DRIVER commit "${BUILDAH_ARGS[@]}" $container $IMAGE
+          set +x
+          echo "######################################################$TLSVERIFY####################################"
+          echo "Pushing to ${IMAGE%:*}:${TASKRUN_NAME}"
+          echo "##########################################################################################"
+          if ! retry buildah --storage-driver $STORAGE_DRIVER push \
+            --tls-verify="$TLSVERIFY" \
+            "$IMAGE" \
+            "docker://${IMAGE%:*}:$(context.taskRun.name)"; then
+            echo "Failed to push sbom image to ${IMAGE%:*}:$(context.taskRun.name) after ${max_run} tries"
+            exit 1
+          fi
+          
+          echo "##########################################################################################"
+          echo "Pushing to ${IMAGE}"
+          echo "##########################################################################################"
+          if ! retry buildah --storage-driver $STORAGE_DRIVER push \
```

Comment:
We can simplify retry using buildah parameter directly as was done here https://github.com/konflux-ci/build-definitions/pull/1663

Reply:
Logic adapted and reviewed. See: c9f05819